base/baseext.jl

```diff
@@ -2,6 +2,15 @@
 
 # extensions to Core types to add features in Base
 
+"""
+    VecElement{T}
+
+Element type of vectors with elements of type `T`. Homogeneous tuples with
```

Comment:
> Element type of vectors with elements of type `T`.

I think this is too vague to make sense. Out of context, it can easily be interpreted as "`eltype` of  vectors (of type `AbstractVector`) that have `eltype` equal to `T`", that is `T`, so `VecElement{T} == T`...

Reply:
Right. This is a kind of self-reference. Does the following make more sense?

> A wrapper type of a SIMD (single instruction, multiple data) vector's
elements of type `T` (i.e., homogeneous tuples with this type element
represent a SIMD vector, such as `NTuple{4,VecElement{Float32}}`).